Add Source Gen Logging for Worker Config

aishwaryabh opened the pull request:
<!-- Please provide all the information below.  -->

### Issue describing the changes in this PR

resolves [#9503](https://github.com/Azure/azure-functions-host/issues/9503)

This PR adds logging for the worker config using json source gen. I also checked in the benchmark tests I created for reference

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

<!-- Optional: delete if not applicable  -->
### Additional information

Additional PR information

aishwaryabh committed:
changes

aishwaryabh committed:
changes

brettsam commented on src/WebJobs.Script/Workers/Rpc/Configuration/LanguageWorkerOptions.cs:
```diff
@@ -1,12 +1,25 @@
+
+    [JsonSourceGenerationOptions(WriteIndented = true, IncludeFields = false)]
+    [JsonSerializable(typeof(LanguageWorkerOptions))]
+    internal partial class SourceGenerationContextWorkerConfig : JsonSerializerContext
```
Let's call this `LanguageWorkerOptionsJsonSerializerContext`...

brettsam replied:
Oh and you can remove `IncludeFields = false` -- I did that as a test earlier.

brettsam commented on src/WebJobs.Script/WebJobs.Script.csproj:
```diff
@@ -83,6 +83,7 @@
     <PackageReference Include="System.Reactive.Linq" Version="5.0.0" />
     <PackageReference Include="System.Reactive.Core" Version="5.0.0" />
     <PackageReference Include="System.Runtime.Loader" Version="4.3.0" />
+    <PackageReference Include="System.Text.Json" Version="6.0.9" />
```
Can you add a comment with a note here? Something like:

```xml
<!--
Explicitly referencing System.Text.Json to get two bug fixes for the Source Generator. Without this, the 6.0.0 version 
of the Source Generator is used, which causes issues. This reference can be removed in future framework versions.

(and then add links to those two issues?) 
-->
```

aishwaryabh replied:
Added!

brettsam commented on benchmarks/WebJobs.Script.Benchmarks/SerializationBenchmarks.cs:
```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```
You can back this out... we don't need it checked in.

aishwaryabh replied:
Removed!

aishwaryabh committed:
addressing comments

brettsam commented on src/WebJobs.Script/WebJobs.Script.csproj:
```diff
@@ -83,6 +83,14 @@
+    of the Source Generator is used, which causes issues. This reference can be removed in future framework versions.
+
+    https://github.com/dotnet/runtime/issues/62354
+	https://github.com/dotnet/runtime/issues/62354
```
Can you fix the spacing? looks off on the whole comment. Also one of these should be issue 62082

aishwaryabh commented:
deps.json unit test:

```
 Error Message:
   IMPORTANT: The dependencies in WebHost have changed and MUST be reviewed before proceeding. Please follow up with brettsam, fabiocav or mathewc for approval.

Previous file: D:\a\_work\1\s\test\WebJobs.Script.Tests\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json
New file:      D:\a\_work\1\s\src\WebJobs.Script.WebHost\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json

  Changed:

  Removed:

  Added:
    - System.Text.Json.dll: 6.0.0.0/6.0.2523.51912

```

aishwaryabh committed:
updating deps.json

aishwaryabh committed:
versioning

aishwaryabh committed:
logging changes

aishwaryabh committed:
changes
